diff --git a/dependencyManagement/build.gradle.kts b/dependencyManagement/build.gradle.kts
index 98def282f8..65fd6a8a13 100644
--- a/dependencyManagement/build.gradle.kts
+++ b/dependencyManagement/build.gradle.kts
@@ -104,7 +104,7 @@ val DEPENDENCIES = listOf(
   "io.netty:netty:3.10.6.Final",
   "io.opentelemetry.contrib:opentelemetry-azure-resources:${otelContribVersion}",
   "io.opentelemetry.contrib:opentelemetry-aws-resources:${otelContribVersion}",
-  "io.opentelemetry.contrib:opentelemetry-aws-xray-propagator:${otelContribVersion}",
+  "io.opentelemetry.contrib:opentelemetry-aws-xray-propagator:1.48.0-alpha-adot1",
   "io.opentelemetry.contrib:opentelemetry-gcp-resources:${otelContribVersion}",
   "io.opentelemetry.contrib:opentelemetry-cloudfoundry-resources:${otelContribVersion}",
   "io.opentelemetry.contrib:opentelemetry-baggage-processor:${otelContribVersion}",
diff --git a/instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/TomcatIntegrationTest.java b/instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/TomcatIntegrationTest.java
index 1234567890..abcdef1234 100644
--- a/instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/TomcatIntegrationTest.java
+++ b/instrumentation/jmx-metrics/library/src/test/java/io/opentelemetry/instrumentation/jmx/rules/TomcatIntegrationTest.java
@@ -21,7 +21,6 @@ public class TomcatIntegrationTest extends TargetSystemTest {
   @ParameterizedTest
   @CsvSource({
     "tomcat:10.0, https://tomcat.apache.org/tomcat-10.0-doc/appdev/sample/sample.war",
-    "tomcat:9.0, https://tomcat.apache.org/tomcat-9.0-doc/appdev/sample/sample.war"
   })
   void testCollectedMetrics(String dockerImageName, String sampleWebApplicationUrl)
       throws Exception {
diff --git a/version.gradle.kts b/version.gradle.kts
index 023d04703c..ec9690086c 100644
--- a/version.gradle.kts
+++ b/version.gradle.kts
@@ -1,5 +1,5 @@
-val stableVersion = "2.18.1"
-val alphaVersion = "2.18.1-alpha"
+val stableVersion = "2.18.1-adot1"
+val alphaVersion = "2.18.1-adot1-alpha"
 
 allprojects {
   if (findProperty("otel.stable") != "true") {
-- 
2.45.1

