diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
index f2ab7c2d..7dcf8347 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
@@ -263,7 +263,7 @@ final class XrayRulesSampler implements Sampler {
           ruleToReportTo = applier;
           break;
         }
-        if (applier.matches(spanData.getAttributes(), resource)) {
+        if (matchedRule == null && applier.matches(spanData.getAttributes(), resource)) {
           matchedRule = applier;
         }
       }
diff --git a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
index c8a8dead..be314c1c 100644
--- a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
+++ b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
@@ -571,11 +571,27 @@ class XrayRulesSamplerTest {
             0.0,
             "*",
             "*",
-            4,
+            2,
             0,
             "*",
             "*",
-            "default-rule",
+            "applied-boost-rule",
+            "*",
+            "*",
+            "*",
+            1,
+            SamplingRateBoost.create(1, 300));
+    SamplingRule rule3 =
+        SamplingRule.create(
+            Collections.emptyMap(),
+            0.0,
+            "*",
+            "*",
+            3,
+            0,
+            "*",
+            "*",
+            "lower-priority-boost-rule",
             "*",
             "*",
             "*",
@@ -603,7 +619,7 @@ class XrayRulesSamplerTest {
             Resource.getDefault(),
             clock,
             Sampler.alwaysOn(),
-            Arrays.asList(rule1, rule2),
+            Arrays.asList(rule1, rule2, rule3),
             config);

     Instant now = Instant.ofEpochSecond(0, clock.now());
@@ -640,13 +656,16 @@ class XrayRulesSamplerTest {
     // Rules are ordered by priority, so cat-rule is first
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
-
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getTotalCount()).isEqualTo(3);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(3);
-
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);

+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+
     // Mock trace coming from upstream service where it was sampled by cat-rule
     when(spanDataMock.getTraceId()).thenReturn("TRACE_ID4");
     when(readableSpanMock.getSpanContext())
@@ -671,6 +690,9 @@ class XrayRulesSamplerTest {
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);

     // Assert the trace ID cache is filled with appropriate data and is cleared after TTL passes
     assertThat(sampler.getTraceUsageCache().asMap().size()).isEqualTo(4);
diff --git a/version.gradle.kts b/version.gradle.kts
index 74967704..d08094f7 100644
--- a/version.gradle.kts
+++ b/version.gradle.kts
@@ -1,5 +1,5 @@
-val stableVersion = "1.52.0"
-val alphaVersion = "1.52.0-alpha"
+val stableVersion = "1.52.0-adot1"
+val alphaVersion = "1.52.0-alpha-adot1"
 val tagVersion by extra { "v$stableVersion" }

 allprojects {
-- 
2.45.1

