diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSampler.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSampler.java
index bfb2eee6..d9adedfe 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSampler.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSampler.java
@@ -235,7 +235,8 @@ public final class AwsXrayRemoteSampler implements Sampler, Closeable {
       statisticsSnapshot.stream()
           .forEach(
               snapshot -> {
-                if (snapshot.getStatisticsDocument() != null) {
+                if (snapshot.getStatisticsDocument() != null
+                    && snapshot.getStatisticsDocument().getRequestCount() > 0) {
                   statistics.add(snapshot.getStatisticsDocument());
                 }
                 if (snapshot.getBoostStatisticsDocument() != null
@@ -243,6 +244,12 @@ public final class AwsXrayRemoteSampler implements Sampler, Closeable {
                   boostStatistics.add(snapshot.getBoostStatisticsDocument());
                 }
               });
+
+      if (statistics.isEmpty() && boostStatistics.isEmpty()) {
+        fetchTargetsFuture =
+            executor.schedule(this::fetchTargets, DEFAULT_TARGET_INTERVAL_NANOS, NANOSECONDS);
+        return;
+      }
       Set<String> requestedTargetRuleNames =
           statistics.stream().map(SamplingStatisticsDocument::getRuleName).collect(toSet());

diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/GetSamplingRulesResponse.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/GetSamplingRulesResponse.java
index 01835dc2..962a4602 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/GetSamplingRulesResponse.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/GetSamplingRulesResponse.java
@@ -118,13 +118,18 @@ abstract class GetSamplingRulesResponse {
     @JsonCreator
     static SamplingRateBoost create(
         @JsonProperty("MaxRate") double maxRate,
-        @JsonProperty("CooldownWindowMinutes") long cooldownWindowMinutes) {
+        @JsonProperty("CooldownWindowMinutes") long cooldownWindowMinutes,
+        @JsonProperty("DisableDefaultAnomalyDetection") @Nullable
+            Boolean disableDefaultAnomalyDetection) {
       return new AutoValue_GetSamplingRulesResponse_SamplingRateBoost(
-          maxRate, cooldownWindowMinutes);
+          maxRate, cooldownWindowMinutes, disableDefaultAnomalyDetection);
     }

     abstract double getMaxRate();

     abstract long getCooldownWindowMinutes();
+
+    @Nullable
+    abstract Boolean getDisableDefaultAnomalyDetection();
   }
 }
diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java
index 0e78d5f5..792812eb 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java
@@ -15,6 +15,7 @@ import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.api.trace.SpanKind;
 import io.opentelemetry.context.Context;
+import io.opentelemetry.contrib.awsxray.GetSamplingRulesResponse.SamplingRateBoost;
 import io.opentelemetry.contrib.awsxray.GetSamplingTargetsRequest.SamplingBoostStatisticsDocument;
 import io.opentelemetry.contrib.awsxray.GetSamplingTargetsRequest.SamplingStatisticsDocument;
 import io.opentelemetry.contrib.awsxray.GetSamplingTargetsResponse.SamplingBoost;
@@ -89,6 +90,7 @@ final class SamplingRuleApplier {

   // Adaptive sampling related configs
   private final boolean hasBoost;
+  private final boolean defaultAnomalyDetectionDisabled;
   private final double boostedFixedRate;
   private final Long boostEndTimeNanos;
   private final Sampler boostedFixedRateSampler;
@@ -145,7 +147,12 @@ final class SamplingRuleApplier {
     fixedRateSampler = createFixedRate(fixedRate);

     // Check if the rule has a sampling rate boost option
-    hasBoost = rule.getSamplingRateBoost() != null;
+    SamplingRateBoost samplingRateBoost = rule.getSamplingRateBoost();
+    hasBoost = samplingRateBoost != null;
+
+    defaultAnomalyDetectionDisabled =
+        samplingRateBoost != null
+            && Boolean.TRUE.equals(samplingRateBoost.getDisableDefaultAnomalyDetection());

     boostedFixedRate = fixedRate;
     boostedFixedRateSampler = createFixedRate(fixedRate);
@@ -182,6 +189,7 @@ final class SamplingRuleApplier {
       double boostedFixedRate,
       Long boostEndTimeNanos,
       boolean hasBoost,
+      boolean defaultAnomalyDetectionDisabled,
       Map<String, Matcher> attributeMatchers,
       Matcher urlPathMatcher,
       Matcher serviceNameMatcher,
@@ -203,6 +211,7 @@ final class SamplingRuleApplier {
     this.boostedFixedRate = boostedFixedRate;
     this.boostEndTimeNanos = boostEndTimeNanos;
     this.hasBoost = hasBoost;
+    this.defaultAnomalyDetectionDisabled = defaultAnomalyDetectionDisabled;
     this.attributeMatchers = attributeMatchers;
     this.urlPathMatcher = urlPathMatcher;
     this.serviceNameMatcher = serviceNameMatcher;
@@ -431,6 +440,7 @@ final class SamplingRuleApplier {
         newBoostedFixedRate,
         newBoostEndTimeNanos,
         hasBoost,
+        defaultAnomalyDetectionDisabled,
         attributeMatchers,
         urlPathMatcher,
         serviceNameMatcher,
@@ -456,6 +466,7 @@ final class SamplingRuleApplier {
         boostedFixedRate,
         boostEndTimeNanos,
         hasBoost,
+        defaultAnomalyDetectionDisabled,
         attributeMatchers,
         urlPathMatcher,
         serviceNameMatcher,
@@ -480,6 +491,10 @@ final class SamplingRuleApplier {
     return hasBoost;
   }

+  boolean isDefaultAnomalyDetectionDisabled() {
+    return defaultAnomalyDetectionDisabled;
+  }
+
   @Nullable
   private static String getArn(Attributes attributes, Resource resource) {
     String arn = resource.getAttributes().get(AWS_ECS_CONTAINER_ARN);
@@ -640,8 +655,6 @@ final class SamplingRuleApplier {
     final SamplingStatisticsDocument statisticsDocument;
     final SamplingBoostStatisticsDocument boostStatisticsDocument;

-    // final SamplingBoostStatisticsDocument boostStatisticsDocument;
-
     SamplingRuleStatisticsSnapshot(
         SamplingStatisticsDocument statisticsDocument,
         SamplingBoostStatisticsDocument boostStatisticsDocument) {
diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
index f2ab7c2d..6f870eae 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java
@@ -50,6 +50,8 @@ final class XrayRulesSampler implements Sampler {

   public static final AttributeKey<String> AWS_XRAY_SAMPLING_RULE =
       AttributeKey.stringKey("aws.xray.sampling_rule");
+  public static final AttributeKey<Boolean> AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED =
+      AttributeKey.booleanKey("aws.xray.adaptive_sampling_configured");

   // Used for generating operation
   private static final String UNKNOWN_OPERATION = "UnknownOperation";
@@ -183,6 +185,9 @@ final class XrayRulesSampler implements Sampler {
                 .put(
                     AWS_XRAY_SAMPLING_RULE.getKey(),
                     ruleToPropagate != null ? ruleToPropagate : "UNKNOWN")
+                .put(
+                    AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED.getKey(),
+                    this.adaptiveSamplingConfig != null)
                 .build(),
             hashedRule);
       }
@@ -225,7 +230,9 @@ final class XrayRulesSampler implements Sampler {
       return;
     }

-    AnomalyDetectionResult result = isAnomaly(span, spanData);
+    SamplingRuleApplier effectiveRule = getEffectiveSamplingRuleApplier(span, spanData);
+
+    AnomalyDetectionResult result = isAnomaly(span, spanData, effectiveRule);
     boolean shouldBoostSampling = result.shouldBoostSampling();
     boolean shouldCaptureAnomalySpan = result.shouldCaptureAnomalySpan();

@@ -247,52 +254,57 @@ final class XrayRulesSampler implements Sampler {
     // Sampling Boost
     boolean isCountedAsAnomalyForBoost = false;
     if (shouldBoostSampling || isNewTrace) {
-      String traceStateValue =
-          span.getSpanContext()
-              .getTraceState()
-              .get(AwsSamplingResult.AWS_XRAY_SAMPLING_RULE_TRACE_STATE_KEY);
-      String upstreamRuleName =
-          traceStateValue != null
-              ? hashToRuleMap.getOrDefault(traceStateValue, traceStateValue)
-              : traceStateValue;
-      SamplingRuleApplier ruleToReportTo = null;
-      SamplingRuleApplier matchedRule = null;
-      for (SamplingRuleApplier applier : ruleAppliers) {
-        // Rule propagated from when sampling decision was made, otherwise the matched rule
-        if (applier.getRuleName().equals(upstreamRuleName)) {
-          ruleToReportTo = applier;
-          break;
-        }
-        if (applier.matches(spanData.getAttributes(), resource)) {
-          matchedRule = applier;
-        }
-      }
-      if (ruleToReportTo == null) {
-        if (matchedRule == null) {
-          logger.log(
-              FINE,
-              "No sampling rule matched the request. This is a bug in either the OpenTelemetry SDK or X-Ray.");
-        } else if (!span.getParentSpanContext().isValid()) {
-          // Span is not from an upstream service, so we should boost the matched rule
-          ruleToReportTo = matchedRule;
-        }
+      if (effectiveRule == null) {
+        logger.log(
+            FINE,
+            "No sampling rule matched the request. This is a bug in either the OpenTelemetry SDK or X-Ray.");
       }

       if (shouldBoostSampling
-          && ruleToReportTo != null
-          && ruleToReportTo.hasBoost()
+          && effectiveRule != null
+          && effectiveRule.hasBoost()
           && !AwsXrayAdaptiveSamplingConfig.UsageType.isUsedForBoost(existingUsage)) {
-        ruleToReportTo.countAnomalyTrace(span);
+        effectiveRule.countAnomalyTrace(span);
         isCountedAsAnomalyForBoost = true;
       }
-      if (isNewTrace && ruleToReportTo != null && ruleToReportTo.hasBoost()) {
-        ruleToReportTo.countTrace();
+      if (isNewTrace && effectiveRule != null && effectiveRule.hasBoost()) {
+        effectiveRule.countTrace();
       }
     }

     updateTraceUsageCache(traceId, isSpanCaptured, isCountedAsAnomalyForBoost);
   }

+  @Nullable
+  private SamplingRuleApplier getEffectiveSamplingRuleApplier(
+      ReadableSpan span, SpanData spanData) {
+    String traceStateValue =
+        span.getSpanContext()
+            .getTraceState()
+            .get(AwsSamplingResult.AWS_XRAY_SAMPLING_RULE_TRACE_STATE_KEY);
+    String upstreamRuleName =
+        traceStateValue != null
+            ? hashToRuleMap.getOrDefault(traceStateValue, traceStateValue)
+            : null;
+
+    SamplingRuleApplier matchedRule = null;
+    for (SamplingRuleApplier applier : ruleAppliers) {
+      if (applier.getRuleName().equals(upstreamRuleName)) {
+        return applier;
+      }
+      if (matchedRule == null && applier.matches(spanData.getAttributes(), resource)) {
+        matchedRule = applier;
+      }
+    }
+
+    // No upstream rule found, use matched rule for root spans
+    if (matchedRule != null && !span.getParentSpanContext().isValid()) {
+      return matchedRule;
+    }
+
+    return null;
+  }
+
   List<SamplingRuleApplier.SamplingRuleStatisticsSnapshot> snapshot(Date now) {
     return Arrays.stream(ruleAppliers)
         .map(rule -> rule.snapshot(now))
@@ -323,14 +335,8 @@ final class XrayRulesSampler implements Sampler {
                   if (target != null) {
                     return rule.withTarget(target, now, currentNanoTime);
                   }
-                  if (requestedTargetRuleNames.contains(rule.getRuleName())) {
-                    // In practice X-Ray should return a target for any rule we requested but
-                    // do a defensive check here in case. If we requested a target but got nothing
-                    // back assume the default interval.
-                    return rule.withNextSnapshotTimeNanos(defaultNextSnapshotTimeNanos);
-                  }
                   // Target not requested, will be updated in a future target fetch.
-                  return rule;
+                  return rule.withNextSnapshotTimeNanos(defaultNextSnapshotTimeNanos);
                 })
             .toArray(SamplingRuleApplier[]::new);
     return new XrayRulesSampler(
@@ -345,7 +351,8 @@ final class XrayRulesSampler implements Sampler {
         traceUsageCache);
   }

-  private AnomalyDetectionResult isAnomaly(ReadableSpan span, SpanData spanData) {
+  private AnomalyDetectionResult isAnomaly(
+      ReadableSpan span, SpanData spanData, @Nullable SamplingRuleApplier effectiveRule) {
     boolean shouldBoostSampling = false;
     boolean shouldCaptureAnomalySpan = false;
     Long statusCode = spanData.getAttributes().get(HTTP_RESPONSE_STATUS_CODE);
@@ -417,8 +424,13 @@ final class XrayRulesSampler implements Sampler {
         || (statusCode == null
             && spanData.getStatus() != null
             && StatusCode.ERROR.equals(spanData.getStatus().getStatusCode()))) {
-      shouldBoostSampling = true;
-      shouldCaptureAnomalySpan = true;
+      // Default anomaly detection - check if disabled for the effective rule
+      boolean defaultAnomalyDetectionDisabled =
+          effectiveRule != null && effectiveRule.isDefaultAnomalyDetectionDisabled();
+      if (!defaultAnomalyDetectionDisabled) {
+        shouldBoostSampling = true;
+        shouldCaptureAnomalySpan = true;
+      }
     }

     return new AnomalyDetectionResult(shouldBoostSampling, shouldCaptureAnomalySpan);
diff --git a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XraySamplerClient.java b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XraySamplerClient.java
index ef47ba72..f7b60eea 100644
--- a/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XraySamplerClient.java
+++ b/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XraySamplerClient.java
@@ -51,7 +51,7 @@ final class XraySamplerClient {

   private static final ObjectMapper OBJECT_MAPPER =
       new ObjectMapper()
-          .setDefaultPropertyInclusion(JsonInclude.Include.NON_EMPTY)
+          .setDefaultPropertyInclusion(JsonInclude.Include.NON_NULL)
           // AWS APIs return timestamps as floats.
           .registerModule(
               new SimpleModule().addDeserializer(Date.class, new FloatDateDeserializer()))
diff --git a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSamplerTest.java b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSamplerTest.java
index 7a75f377..8ce009c0 100644
--- a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSamplerTest.java
+++ b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSamplerTest.java
@@ -33,6 +33,7 @@ import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.time.Duration;
 import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
@@ -69,6 +70,7 @@ class AwsXrayRemoteSamplerTest {

   private static final AtomicReference<byte[]> rulesResponse = new AtomicReference<>();
   private static final AtomicReference<byte[]> targetsResponse = new AtomicReference<>();
+  private static final AtomicInteger targetsRequestCount = new AtomicInteger(0);

   private static final String TRACE_ID = TraceId.fromLongs(1, 2);

@@ -92,6 +94,7 @@ class AwsXrayRemoteSamplerTest {
           sb.service(
               "/SamplingTargets",
               (ctx, req) -> {
+                targetsRequestCount.incrementAndGet();
                 byte[] response = AwsXrayRemoteSamplerTest.targetsResponse.get();
                 if (response == null) {
                   // Error out until the test configures a response, the sampler will use the
@@ -120,6 +123,7 @@ class AwsXrayRemoteSamplerTest {
   void tearDown() {
     sampler.close();
     rulesResponse.set(null);
+    targetsRequestCount.set(0);
   }

   @Test
@@ -163,6 +167,26 @@ class AwsXrayRemoteSamplerTest {
             });
   }

+  @Test
+  void doesNotCallGetSamplingTargetsWhenNoStatistics() {
+    rulesResponse.set(RULE_RESPONSE_1);
+    targetsResponse.set(TARGETS_RESPONSE);
+
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(sampler.getDescription()).contains("XrayRulesSampler");
+            });
+
+    await()
+        .pollDelay(Duration.ofMillis(100))
+        .atMost(Duration.ofMillis(200))
+        .untilAsserted(
+            () -> {
+              assertThat(targetsRequestCount.get()).isEqualTo(0);
+            });
+  }
+
   @Test
   void defaultInitialSampler() {
     try (AwsXrayRemoteSampler sampler = AwsXrayRemoteSampler.newBuilder(Resource.empty()).build()) {
diff --git a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
index c8a8dead..59d9c982 100644
--- a/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
+++ b/aws-xray/src/test/java/io/opentelemetry/contrib/awsxray/XrayRulesSamplerTest.java
@@ -134,6 +134,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "cat-service"))
@@ -143,6 +144,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -152,6 +154,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -161,6 +164,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -170,6 +174,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -179,6 +184,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));
     assertThat(doSample(sampler, "unknown"))
@@ -188,6 +194,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "default-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("default-rule")));

@@ -222,6 +229,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -231,6 +239,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "unknown"))
@@ -240,6 +249,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "default-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("default-rule")));
     // Targets overridden to always drop.
@@ -250,6 +260,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -259,6 +270,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, false)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));

@@ -367,6 +379,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "cat-service"))
@@ -376,6 +389,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -385,6 +399,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -394,6 +409,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -403,6 +419,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -412,6 +429,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));
     assertThat(doSample(sampler, "unknown"))
@@ -421,6 +439,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "default-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("default-rule")));

@@ -455,6 +474,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.RECORD_AND_SAMPLE,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "dog-service"))
@@ -464,6 +484,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "dog-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("dog-rule")));
     assertThat(doSample(sampler, "unknown"))
@@ -473,6 +494,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "default-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("default-rule")));
     // Targets overridden to always drop.
@@ -483,6 +505,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "cat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("cat-rule")));
     assertThat(doSample(sampler, "bat-service"))
@@ -492,6 +515,7 @@ class XrayRulesSamplerTest {
                 SamplingDecision.DROP,
                 Attributes.builder()
                     .put(XrayRulesSampler.AWS_XRAY_SAMPLING_RULE, "bat-rule")
+                    .put(XrayRulesSampler.AWS_XRAY_ADAPTIVE_SAMPLING_CONFIGURED, true)
                     .build(),
                 XrayRulesSampler.hashRuleName("bat-rule")));

@@ -571,16 +595,32 @@ class XrayRulesSamplerTest {
             0.0,
             "*",
             "*",
-            4,
+            2,
             0,
             "*",
             "*",
-            "default-rule",
+            "applied-boost-rule",
             "*",
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));
+    SamplingRule rule3 =
+        SamplingRule.create(
+            Collections.emptyMap(),
+            0.0,
+            "*",
+            "*",
+            3,
+            0,
+            "*",
+            "*",
+            "lower-priority-boost-rule",
+            "*",
+            "*",
+            "*",
+            1,
+            SamplingRateBoost.create(1, 300, null));
     AwsXrayAdaptiveSamplingConfig config =
         AwsXrayAdaptiveSamplingConfig.builder()
             .setVersion(1.0)
@@ -603,7 +643,7 @@ class XrayRulesSamplerTest {
             Resource.getDefault(),
             clock,
             Sampler.alwaysOn(),
-            Arrays.asList(rule1, rule2),
+            Arrays.asList(rule1, rule2, rule3),
             config);

     Instant now = Instant.ofEpochSecond(0, clock.now());
@@ -640,13 +680,16 @@ class XrayRulesSamplerTest {
     // Rules are ordered by priority, so cat-rule is first
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
-
     assertThat(snapshot.get(0).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getTotalCount()).isEqualTo(3);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(3);
-
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);

+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+
     // Mock trace coming from upstream service where it was sampled by cat-rule
     when(spanDataMock.getTraceId()).thenReturn("TRACE_ID4");
     when(readableSpanMock.getSpanContext())
@@ -671,6 +714,9 @@ class XrayRulesSamplerTest {
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
     assertThat(snapshot.get(1).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getTotalCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getAnomalyCount()).isEqualTo(0);
+    assertThat(snapshot.get(2).getBoostStatisticsDocument().getSampledAnomalyCount()).isEqualTo(0);

     // Assert the trace ID cache is filled with appropriate data and is cleared after TTL passes
     assertThat(sampler.getTraceUsageCache().asMap().size()).isEqualTo(4);
@@ -731,7 +777,7 @@ class XrayRulesSamplerTest {
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));

     TestClock clock = TestClock.create();
     AwsXrayAdaptiveSamplingConfig config =
@@ -803,7 +849,7 @@ class XrayRulesSamplerTest {
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));

     TestClock clock = TestClock.create();
     AwsXrayAdaptiveSamplingConfig config =
@@ -875,7 +921,7 @@ class XrayRulesSamplerTest {
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));

     TestClock clock = TestClock.create();
     AwsXrayAdaptiveSamplingConfig config =
@@ -961,7 +1007,7 @@ class XrayRulesSamplerTest {
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));

     TestClock clock = TestClock.create();
     AwsXrayAdaptiveSamplingConfig config =
@@ -1054,7 +1100,7 @@ class XrayRulesSamplerTest {
             "*",
             "*",
             1,
-            SamplingRateBoost.create(1, 300));
+            SamplingRateBoost.create(1, 300, null));

     TestClock clock = TestClock.create();
     // Error span capture should default to 1/s
diff --git a/aws-xray/src/test/resources/get-sampling-targets-request.json b/aws-xray/src/test/resources/get-sampling-targets-request.json
index a8ad4540..83e32229 100644
--- a/aws-xray/src/test/resources/get-sampling-targets-request.json
+++ b/aws-xray/src/test/resources/get-sampling-targets-request.json
@@ -16,5 +16,6 @@
       "SampledCount":31,
       "BorrowCount":0
     }
-  ]
+  ],
+  "SamplingBoostStatisticsDocuments":[]
 }
diff --git a/version.gradle.kts b/version.gradle.kts
index 74967704..d08094f7 100644
--- a/version.gradle.kts
+++ b/version.gradle.kts
@@ -1,5 +1,5 @@
-val stableVersion = "1.52.0"
-val alphaVersion = "1.52.0-alpha"
+val stableVersion = "1.52.0-adot1"
+val alphaVersion = "1.52.0-alpha-adot1"
 val tagVersion by extra { "v$stableVersion" }

 allprojects {
-- 
2.45.1

