commit 9ad7e335023493544a02128c75f5ba12532a92eb
Author: Bryan Aguilar <bryaag@amazon.com>
Date:   Fri Jul 28 11:08:04 2023 -0700

    Patch instrumentation versions

diff --git a/dependencyManagement/build.gradle.kts b/dependencyManagement/build.gradle.kts
index 39d3c109c2..0caba721a6 100644
--- a/dependencyManagement/build.gradle.kts
+++ b/dependencyManagement/build.gradle.kts
@@ -12,7 +12,7 @@ val dependencyVersions = hashMapOf<String, String>()
 rootProject.extra["versions"] = dependencyVersions
 
 // this line is managed by .github/scripts/update-sdk-version.sh
-val otelSdkVersion = "1.28.0"
+val otelSdkVersion = "1.28.0-adot1"
 val otelSdkAlphaVersion = otelSdkVersion.replaceFirst("(-SNAPSHOT)?$".toRegex(), "-alpha$1")
 
 // Need both BOM and groovy jars
diff --git a/examples/distro/build.gradle b/examples/distro/build.gradle
index 9a97747771..2dd29a055f 100644
--- a/examples/distro/build.gradle
+++ b/examples/distro/build.gradle
@@ -27,7 +27,7 @@ subprojects {
   ext {
     versions = [
       // this line is managed by .github/scripts/update-sdk-version.sh
-      opentelemetrySdk           : "1.28.0",
+      opentelemetrySdk           : "1.28.0-adot1",
 
       // these lines are managed by .github/scripts/update-version.sh
       opentelemetryJavaagent     : "1.28.0",
diff --git a/examples/extension/build.gradle b/examples/extension/build.gradle
index 1c25a1e530..39000803bf 100644
--- a/examples/extension/build.gradle
+++ b/examples/extension/build.gradle
@@ -23,7 +23,7 @@ version '1.0'
 ext {
   versions = [
     // this line is managed by .github/scripts/update-sdk-version.sh
-    opentelemetrySdk           : "1.28.0",
+    opentelemetrySdk           : "1.28.0-adot1",
 
     // these lines are managed by .github/scripts/update-version.sh
     opentelemetryJavaagent     : "1.28.0",
diff --git a/version.gradle.kts b/version.gradle.kts
index cc1414c0bf..c7b1ea682e 100644
--- a/version.gradle.kts
+++ b/version.gradle.kts
@@ -1,5 +1,5 @@
-val stableVersion = "1.28.0"
-val alphaVersion = "1.28.0-alpha"
+val stableVersion = "1.28.0-adot1"
+val alphaVersion = "1.28.0-adot1-alpha"
 
 allprojects {
   if (findProperty("otel.stable") != "true") {
