commit 44ea45fa7b16aa407da199c03841b382ecdc5bd0
Author: Bryan Aguilar <bryaag@amazon.com>
Date:   Fri Jul 28 09:51:48 2023 -0700

    Bump okio-bom and java ver

diff --git a/dependencyManagement/build.gradle.kts b/dependencyManagement/build.gradle.kts
index 917446df7..df564c863 100644
--- a/dependencyManagement/build.gradle.kts
+++ b/dependencyManagement/build.gradle.kts
@@ -17,7 +17,7 @@ val DEPENDENCY_BOMS = listOf(
   "com.google.protobuf:protobuf-bom:3.23.4",
   "com.linecorp.armeria:armeria-bom:1.24.2",
   "com.squareup.okhttp3:okhttp-bom:4.11.0",
-  "com.squareup.okio:okio-bom:3.3.0", // applies to transitive dependencies of okhttp
+  "com.squareup.okio:okio-bom:3.4.0", // applies to transitive dependencies of okhttp
   "io.grpc:grpc-bom:1.56.1",
   "io.netty:netty-bom:4.1.94.Final",
   "io.zipkin.brave:brave-bom:5.16.0",
diff --git a/version.gradle.kts b/version.gradle.kts
index 1e1986816..32fb83632 100644
--- a/version.gradle.kts
+++ b/version.gradle.kts
@@ -1,7 +1,7 @@
 val snapshot = false
 
 allprojects {
-  var ver = "1.28.0"
+  var ver = "1.28.0-adot1"
   val release = findProperty("otel.release")
   if (release != null) {
     ver += "-" + release
